pkg/fqdn/cache.go

```diff
@@ -233,50 +231,52 @@ func (c *DNSCache) addNameToCleanup(entry *cacheEntry) {
 	c.cleanup[expiration] = append(expiredEntries, entry.Name)
 }
 
-// cleanupExpiredEntries cleans all the expired entries from the lastTime that
-// runs to the give time. It will lock the struct until retrieves all the data.
-// It returns the list of names that need to be deleted from the policies.
-func (c *DNSCache) cleanupExpiredEntries(expires time.Time) ([]string, time.Time) {
-	timediff := int(expires.Sub(c.lastCleanup).Seconds())
-	startTime := c.lastCleanup
-	expiredEntries := []string{}
-	for i := 0; i < int(timediff); i++ {
-		expiredTime := c.lastCleanup.Add(time.Second)
-		key := expiredTime.Unix()
-		c.lastCleanup = expiredTime
-		entries, exists := c.cleanup[key]
-		if !exists {
-			continue
+// cleanupExpiredEntries cleans all the expired entries since lastCleanup up to
+// expires, but not including it. lastCleanup is set to expires and later
+// cleanups begin from that time.
+// It returns the list of names that have expired data.
```

Comment:
nit:

```suggestion
// It returns the list of names that have expired data and a map of removed DNS cache entries, keyed by IP.
```

More elaborate types for the return values would be another way to document this, but I think a comment like the above would suffice.